ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java

```diff
@@ -418,13 +418,19 @@ private TransactionValidator getTransactionValidator() {
   private ValidationResultAndAccount validateTransaction(
       final Transaction transaction, final boolean isLocal, final boolean hasPriority) {
 
-    final BlockHeader chainHeadBlockHeader = getChainHeadBlockHeader().orElse(null);
+    // Optimistically get the chain head. getChainHeadBlockHeader() doesn't take any locks,
+    // which might mean that the latest block is still being committed to storage. If this
+    // call fails try the synchronized alternative, and if that fails give up.
+    BlockHeader chainHeadBlockHeader = getChainHeadBlockHeader().orElse(null);
```

Comment:
how about we leave this as it is and have this logic directly in getChainBlockHeader()?

Reply:
Yeah that's a good thought. I'll tidy it up and push a new commit.

Reply:
New commits pushed @garyschulte. I'll mark as auto-merge once you're happy with them.